### PR TITLE
mounter: support to output pre table info in ddl events

### PR DESCRIPTION
**Pull request by leoppro**
### What problem does this PR solve? 
support to output pre table info in ddl events

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: cdc/entry/schema_storage.go [hunk 1]
```diff
@@ -59,6 +56,48 @@ func (s *SingleSchemaSnapshot) HandleDDL(job *timodel.Job) error {
...
+	case timodel.ActionRenameTable, timodel.ActionDropTable, timodel.ActionDropView, timodel.ActionTruncateTable:
+		// get the table will be dropped
+		table, ok := s.TableByID(job.TableID)
+		if !ok {
+			return nil, errors.NotFoundf("table %d", job.TableID)
+		}
+		return table, nil
+	case timodel.ActionTruncateTablePartition, timodel.ActionAddTablePartition, timodel.ActionDropTablePartition:
+		tableID := job.BinlogInfo.TableInfo.ID
+		table, ok := s.TableByID(tableID)
```
Comment (amyangfei) [inline cdc/entry/schema_storage.go:76]: Is it sure to return the table that the partition belongs to, if so, seems the logic is similar to the default branch

File: cdc/entry/schema_storage.go [hunk 2]
```diff
@@ -59,6 +56,48 @@ func (s *SingleSchemaSnapshot) HandleDDL(job *timodel.Job) error {
 	return s.handleDDL(job)
 }
 
+// PreTableInfo returns the table info which with overwrite by the specified job
```
Comment (liuzix) [inline cdc/entry/schema_storage.go:59]:
```suggestion
// PreTableInfo returns the table info which will be overwritten by the specified job
```

File: cdc/model/schema_storage.go [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+	return uniqueKeys
+}
+
+// IsColumnUnique returns whether the column is unique
+func (ti *TableInfo) IsColumnUnique(colID int64) bool {
+	_, exist := ti.uniqueColumns[colID]
+	return exist
+}
+
+// ExistTableUniqueColumn returns whether the table has the unique column
```
Comment (liuzix) [inline cdc/model/schema_storage.go:163]:
```suggestion
// ExistTableUniqueColumn returns whether the table has a unique column
```

**Comment by leoppro** [comment]
@liuzix @amyangfei PTAL again

**Pull request merged**